_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+This guide has been vetted on both Data Hub 5.0 and the Data Hub Framework 4.3.1. 
+
+## Migrating Grove-related artifacts and code to the Data Hub
+
+In order to run a Grove UI project alongside a MarkLogic Data Hub, we recommend migrating Grove related artifacts and code from its code base into your Data Hub project.  
+
+Note that this implies some coordination between the Data Hub project and Grove even after migration of Grove's core capabilities. If possible, it is easiest to manage both projects within the same code repository. The Grove files should all be within a dedicated parent directory. (For example, `grove/ui` and `grove/middle-tier`. Note that there is nothing magic about the `grove` directory name, so call it whatever you want.)
+
+You can accomplish this recommended approach by moving a subset of files from inside the `marklogic` directory of your Grove project into your Data Hub project. Grove projects generated using the grove-cli's `grove new` command ship with this `marklogic` directory, which contains configuration files for a standalone MarkLogic database, optimized for Grove's sample data set.  Once the relevant files are migrated, the `marklogic` directory can be deleted entirely.  Additionally, an environment variable will need to be updated to reflect the data-hub-FINAL HTTP app server as configured by the Data Hub project.
+
+The advantages of this approach include: 
+- Combining the Grove-related artifacts with the Data Hub simplifies development and deployments.  The Grove code is logically separated inside its own directory under `data-hub/src/main/ui-modules` for clarity and distinction from Data Hub specific modules.
+- If code is maintained together in the same repository, consistent versioning can be ensured.
+- The Grove UI becomes a downstream consumer of the Data Hub.
+
+There are some downsides to this approach:
+- UI specific code will reside in close proximity to backend specific configurations.  This might not be as preferable to some development teams.
+- Any CRUD operations performed by the Grove project will write to the Data Hub's FINAL database.
+
+
+### How to implement the recommended approach, step-by-step:
+
+Prerequisite: MarkLogic Data Hub has been deployed to your MarkLogic host.  
+
+[Optional] If you previously installed Grove to its default database via the `mlDeploy` command within the `marklogic` directory, remove the Grove-specific configs before proceeding:
+        
+    > cd grove/marklogic
+    > ./gradlew mlUndeploy
```

Comment:
It is lacking a `-Pconfirm=true`, or would you like users to discover that themselves, as a means to make sure they know what they are doing? ;-)

Reply:
just append " (and confirm)" or something alike..

Reply:
Added the `-Pconfirm=true`